app/components/trainees/school_details/view.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module Trainees
+  module SchoolDetails
+    class View < GovukComponent::Base
+      attr_reader :trainee, :lead_school, :employing_school
+
+      TYPES = {
+        lead: "lead",
+        employing: "employing",
+      }.freeze
+
+      def initialize(trainee)
+        @trainee = trainee
+        @lead_school = trainee.lead_school
+        @employing_school = trainee.employing_school
+      end
+
+      def school_rows
+        [lead_school_row, employing_school_row].compact
+      end
+
+      def lead_school_row
+        return if lead_school.blank?
+
+        {
+          key: t("components.school_details.lead_school_key"),
+          value: school_detail(name(lead_school), location(lead_school)),
+          action: change_link(TYPES[:lead]),
+
+        }
+      end
+
+      def employing_school_row
+        return if employing_school.blank?
+
+        {
+          key: t("components.school_details.employing_school_key"),
+          value: school_detail(name(employing_school), location(employing_school)),
+          action: change_link(TYPES[:employing]),
+        }
+      end
+
+    private
+
+      def school_detail(name, location)
+        tag.p(name, class: "govuk-body") +
+          tag.span(location, class: "govuk-hint")
+      end
+
+      def name(school)
```

Comment:
I was thinking you could get rid of this method altogether and have the following above it:

```ruby
def school_detail(school)
  tag.p(school.name, class: "govuk-body") + tag.span(school_location(school), class: "govuk-hint")
end
```

Reply:
done